src/state-calculator.tsx

```diff
@@ -429,16 +304,160 @@ export const StateCalculator: FC<{
+  ownerStatus: OwnerStatus = DEFAULT_OWNER_STATUS;
+  householdIncome: string = DEFAULT_HOUSEHOLD_INCOME;
+  taxFiling: FilingStatus = DEFAULT_TAX_FILING;
+  householdSize: string = DEFAULT_HOUSEHOLD_SIZE;
+
+  /* attributeChangedCallback() will be called when any of these changes */
+  static observedAttributes = [
+    'lang',
+    'show-email',
+    'include-beta-states',
+    'api-key',
+    'api-host',
+    'state',
+    'zip',
+    'owner-status',
+    'household-income',
+    'tax-filing',
+    'household-size',
+  ] as const;
+
+  reactRootCalculator: Root;
+  reactRootFooter: Root;
+
+  constructor() {
+    super();
+
+    const shadowRoot = this.attachShadow({ mode: 'open' });
+
+    const style = document.createElement('style');
+    style.textContent = tailwindStyles;
+
+    shadowRoot.appendChild(style);
+
+    const calculator = document.createElement('div');
+    shadowRoot.appendChild(calculator);
+    this.reactRootCalculator = createRoot(calculator);
+
+    const footer = document.createElement('div');
+    shadowRoot.appendChild(footer);
+    this.reactRootFooter = createRoot(footer);
+  }
+
+  connectedCallback() {
+    this.render();
+  }
+
+  attributeChangedCallback(
+    attr: (typeof CalculatorElement.observedAttributes)[number],
+    _: string | null, // old value; unused
+    newValue: string | null,
+  ) {
+    if (attr === 'lang') {
+      // Don't do anything; the property already reflects the change.
+      // Assigning to lang here will cause infinite recursion.
+    } else if (attr === 'api-host') {
+      this.apiHost = newValue ?? DEFAULT_CALCULATOR_API_HOST;
+    } else if (attr === 'api-key') {
+      this.apiKey = newValue ?? '';
+    } else if (attr === 'include-beta-states') {
+      this.includeBetaStates = newValue !== null;
+    } else if (attr === 'show-email') {
+      this.showEmail = newValue !== null;
+    } else if (attr === 'state') {
+      this.state = newValue ?? '';
+    } else if (attr === 'zip') {
+      this.zip = newValue ?? DEFAULT_ZIP;
+    } else if (attr === 'owner-status') {
+      this.ownerStatus = (newValue as OwnerStatus) ?? DEFAULT_OWNER_STATUS;
+    } else if (attr === 'household-income') {
+      this.householdIncome = newValue ?? DEFAULT_HOUSEHOLD_INCOME;
+    } else if (attr === 'household-size') {
+      this.householdSize = newValue ?? DEFAULT_HOUSEHOLD_SIZE;
+    } else if (attr === 'tax-filing') {
+      this.taxFiling = (newValue as FilingStatus) ?? DEFAULT_TAX_FILING;
+    } else {
+      // This will fail typechecking if the cases above aren't exhaustive
+      // with respect to observedAttributes
+      const n: never = attr;
+      console.error('Unexpected attribute', n, newValue);
+    }
+    if (this.isConnected) {
+      this.render();
+    }
+  }
+
+  private render() {
+    this.reactRootCalculator.render(
+      <LocaleContext.Provider value={this.lang}>
+        <StateCalculator
+          shadowRoot={this.shadowRoot!}
+          language={this.lang}
```

Comment:
Do we need lang here if it's coming from a context?

Reply:
Yeah, because we need to pass it to the API.

Reply:
Although I suppose the API callsite could take it from React context directly?